x-pack/plugins/security_solution/server/endpoint/routes/actions/state.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RequestHandler } from '@kbn/core/server';
+import { ACTION_STATE_ROUTE } from '../../../../common/endpoint/constants';
+import type {
+  SecuritySolutionPluginRouter,
+  SecuritySolutionRequestHandlerContext,
+} from '../../../types';
+import type { EndpointAppContext } from '../../types';
+import { withEndpointAuthz } from '../with_endpoint_authz';
+
+/**
+ * Registers routes for checking state of actions routes
+ */
+export function registerActionStateRoutes(
+  router: SecuritySolutionPluginRouter,
+  endpointContext: EndpointAppContext,
+  canEncrypt?: boolean
+) {
+  router.get(
+    {
+      path: ACTION_STATE_ROUTE,
+      validate: false,
+      options: { authRequired: true, tags: ['access:securitySolution'] },
+    },
+    withEndpointAuthz(
+      {
+        any: [
```

Comment:
we are likely going to forget to update this as we add more response actions.

Perhaps for a subsequent PR, we should look to see if we can create a `const` that keeps the list of response actions names to required Authz property. This file has something like that defined, but would have to be refactored in order to be re-used:

https://github.com/elastic/kibana/blob/ef05b2eb3a37de364d495814885711a7bc6fecf1/x-pack/plugins/security_solution/public/management/components/endpoint_responder/lib/console_commands_definition.ts#L67

We also have a bunch of "maps" here that could probably be consolidated to also maybe provide this info.: 
https://github.com/elastic/kibana/blob/b7807c5a488197d88a07fbe39de92ccf1e345218/x-pack/plugins/security_solution/common/endpoint/service/response_actions/constants.ts#L68